src/material/list/selection-list.ts

```diff
@@ -385,13 +386,22 @@ export class MatSelectionList extends _MatSelectionListMixinBase implements Focu
       .skipPredicate(() => false)
       .withAllowedModifierKeys(['shiftKey']);
 
-    if (this._tempValues) {
-      this._setOptionsFromValues(this._tempValues);
-      this._tempValues = null;
+    if (this._value) {
+      this._setOptionsFromValues(this._value);
     }
 
+    this.options.changes.pipe(takeUntil(this._destroyed)).subscribe(() => {
+      if (this._value) {
+        // Needs to be deferred, because we're changing bindings as the result of a query
+        // change which results in a "Changed after checked error".
+        Promise.resolve().then(() => {
```

Comment:
Could the option deal with this itself? It will be able to read its parent's state without needing the extra change detection cycle